compiler/rustc_resolve/src/late.rs

```diff
@@ -3146,6 +3189,23 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
         })
     }
 
+    fn resolve_delegation(&mut self, delegation: &'ast Delegation, with_path: bool) {
+        self.in_delegation_item = true;
+
+        let (qself, path) = &delegation.path;
+        if let Some(qself) = &delegation.path.0 {
+            self.visit_ty(&qself.ty);
```

Comment:
Need to do full recursive visiting like this (copypasted from visit.rs):
```rust
            if let Some(qself) = maybe_qself {
                visitor.visit_ty(&qself.ty);
            }
            visitor.visit_path(path, expression.id)
```
The path may have generic arguments, etc, which also needs to be resolved.

Also, recursive visiting typically happens after the current node visiting, `smart_resolve_path` in this case.

Reply:
>Also, recursive visiting typically happens after the current node visiting, `smart_resolve_path` in this case.

^